@preghenella the CI errors are genuine...
````
/mnt/mesos/sandbox/sandbox/sw/SOURCES/O2/4288/0/Detectors/TOF/workflow/include/TOFWorkflow/CompressedAnalysis.h:35:3: warning: annotate this function with 'override' or (rarely) 'final' [modernize-use-override]
  ~CompressedAnalysis() = default;
  ^
````

--- a/Detectors/TOF/compression/CMakeLists.txt
+++ b/Detectors/TOF/compression/CMakeLists.txt
@@ -27,4 +27,10 @@ o2_add_executable(compressed-inspector
                   PUBLIC_LINK_LIBRARIES O2::TOFWorkflowUtils
 		  )
 
+o2_add_executable(compressed-analysis
+                  COMPONENT_NAME tof
+                  SOURCES src/tof-compressed-analysis.cxx
+                  PUBLIC_LINK_LIBRARIES O2::TOFWorkflowUtils
+		  )
+
 

--- a/Detectors/TOF/compression/src/tof-compressed-analysis.cxx
+++ b/Detectors/TOF/compression/src/tof-compressed-analysis.cxx
@@ -1,0 +1,43 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   tof-compressor.cxx
+/// @author Roberto Preghenella
+/// @since  2019-12-18
+/// @brief  Basic DPL workflow for TOF raw data compression
+
+#include "TOFWorkflow/CompressedAnalysisTask.h"
+#include "Framework/WorkflowSpec.h"
+#include "Framework/ConfigParamSpec.h"
+#include "FairLogger.h"
+
+using namespace o2::framework;
+
+// add workflow options, note that customization needs to be declared before
+// including Framework/runDataProcessing
+void customize(std::vector<ConfigParamSpec>& workflowOptions)
+{
+}
+
+#include "Framework/runDataProcessing.h" // the main driver
+
+/// This function hooks up the the workflow specifications into the DPL driver.
+WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
+{
+  return WorkflowSpec{
+    DataProcessorSpec{"compressed-analysis",
+                      select("x:TOF/CRAWDATA"),
+                      Outputs{},
+                      AlgorithmSpec(adaptFromTask<o2::tof::CompressedAnalysisTask>()),
+                      Options{
+                        {"tof-compressed-analysis-conet-mode", VariantType::Bool, false, {"CONET mode"}},
+                        {"tof-compressed-analysis-filename", VariantType::String, "", {"Analysis file name"}},
+                        {"tof-compressed-analysis-function", VariantType::String, "", {"Analysis function call"}}}}};
+}

--- a/Detectors/TOF/workflow/CMakeLists.txt
+++ b/Detectors/TOF/workflow/CMakeLists.txt
@@ -17,6 +17,7 @@ o2_add_library(TOFWorkflowUtils
                        src/TOFRawWriterSpec.cxx
                        src/CompressedDecodingTask.cxx
                        src/CompressedInspectorTask.cxx
+                       src/CompressedAnalysisTask.cxx
                        src/EntropyEncoderSpec.cxx
                        src/EntropyDecoderSpec.cxx
                PUBLIC_LINK_LIBRARIES O2::Framework O2::DPLUtils O2::TOFBase O2::DataFormatsTOF O2::TOFReconstruction)

--- a/Detectors/TOF/workflow/include/TOFWorkflow/CompressedAnalysis.h
+++ b/Detectors/TOF/workflow/include/TOFWorkflow/CompressedAnalysis.h
@@ -1,0 +1,46 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   CompressedAnalysisTask.h
+/// @author Roberto Preghenella
+/// @since  2020-09-04
+/// @brief  TOF compressed data analysis base class
+
+#ifndef O2_TOF_COMPRESSEDANALYSIS
+#define O2_TOF_COMPRESSEDANALYSIS
+
+#include "Headers/RAWDataHeader.h"
+#include "DataFormatsTOF/CompressedDataFormat.h"
+#include "TOFReconstruction/DecoderBase.h"
+
+using namespace o2::tof::compressed;
+
+namespace o2
+{
+namespace tof
+{
+
+class CompressedAnalysis : public DecoderBaseT<o2::header::RAWDataHeaderV6>
+{
+
+ public:
+  CompressedAnalysis() = default;
+  ~CompressedAnalysis() override = default;
+
+  virtual bool initialize() = 0;
+  virtual bool finalize() = 0;
+
+ private:
+};
+
+} // namespace tof
+} // namespace o2
+
+#endif /* O2_TOF_COMPRESSEDANALYSIS */

--- a/Detectors/TOF/workflow/include/TOFWorkflow/CompressedAnalysisTask.h
+++ b/Detectors/TOF/workflow/include/TOFWorkflow/CompressedAnalysisTask.h
@@ -1,0 +1,94 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   CompressedAnalysisTask.h
+/// @author Roberto Preghenella
+/// @since  2020-09-04
+/// @brief  TOF compressed data analysis task
+
+#ifndef O2_TOF_COMPRESSEDANALYSISTASK
+#define O2_TOF_COMPRESSEDANALYSISTASK
+
+#include "Framework/Task.h"
+#include "TOFWorkflow/CompressedAnalysis.h"
+
+#include "TROOT.h"
+#include "TSystem.h"
+#include "TGlobal.h"
+#include "TFunction.h"
+#include <string>
+#include <iostream>
+
+using namespace o2::framework;
+
+namespace o2
+{
+namespace tof
+{
+
+class CompressedAnalysisTask : public Task
+{
+ public:
+  CompressedAnalysisTask() = default;
+  ~CompressedAnalysisTask() override = default;
+  void init(InitContext& ic) final;
+  void run(ProcessingContext& pc) final;
+
+ private:
+  CompressedAnalysis* mAnalysis = nullptr;
+  bool mStatus = false;
+
+  template <typename T>
+  T GetFromMacro(const std::string& file, const std::string& funcname, const std::string& type, const std::string& unique)
+  {
+
+    /** tweak the string to get the required global function **/
+    auto func = funcname;
+    if (func.empty()) {
+      auto size = file.size();
+      auto firstindex = file.find_last_of("/") + 1;
+      auto lastindex = file.find_last_of(".");
+      func = file.substr(firstindex < size ? firstindex : 0,
+                         lastindex < size ? lastindex - firstindex : size - firstindex) +
+             "()";
+    }
+    auto gfunc = func.substr(0, func.find_first_of('('));
+
+    /** load macro is global function is not already defined **/
+    if (!gROOT->GetGlobalFunction(gfunc.c_str())) {
+      if (gROOT->LoadMacro(file.c_str()) != 0) {
+        std::cout << "Cannot find " << file << std::endl;
+        return nullptr;
+      }
+      if (!gROOT->GetGlobalFunction(gfunc.c_str())) {
+        std::cout << "Global function '" << gfunc << "' not defined" << std::endl;
+        return nullptr;
+      }
+    }
+
+    /** check the return type matches the required one **/
+    if (strcmp(gROOT->GetGlobalFunction(gfunc.c_str())->GetReturnTypeName(), type.c_str())) {
+      std::cout << "Global function '" << gfunc << "' does not return a '" << type << "' type" << std::endl;
+      return nullptr;
+    }
+
+    /** process function and retrieve pointer to the returned type **/
+    gROOT->ProcessLine(Form("%s __%s__ = %s;", type.c_str(), unique.c_str(), func.c_str()));
+    auto ptr = (T*)gROOT->GetGlobal(Form("__%s__", unique.c_str()))->GetAddress();
+
+    /** success **/
+    return *ptr;
+  }
+};
+
+} // namespace tof
+} // namespace o2
+
+#endif /* O2_TOF_COMPRESSEDANALYSISTASK */

--- a/Detectors/TOF/workflow/src/CompressedAnalysisTask.cxx
+++ b/Detectors/TOF/workflow/src/CompressedAnalysisTask.cxx
@@ -1,0 +1,98 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   CompressedAnalysisTask.cxx
+/// @author Roberto Preghenella
+/// @since  2020-09-04
+/// @brief  TOF compressed data analysis task
+
+#include "TOFWorkflow/CompressedAnalysisTask.h"
+#include "Framework/Task.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/ControlService.h"
+#include "Framework/CallbackService.h"
+#include "Framework/ConcreteDataMatcher.h"
+#include "Framework/RawDeviceService.h"
+#include "Framework/DeviceSpec.h"
+#include <fairmq/FairMQDevice.h>
+
+using namespace o2::framework;
+
+namespace o2
+{
+namespace tof
+{
+
+void CompressedAnalysisTask::init(InitContext& ic)
+{
+
+  auto conetmode = ic.options().get<bool>("tof-compressed-analysis-conet-mode");
+  auto filename = ic.options().get<std::string>("tof-compressed-analysis-filename");
+  auto function = ic.options().get<std::string>("tof-compressed-analysis-function");
+
+  if (filename.empty()) {
+    LOG(ERROR) << "No analysis filename defined";
+    mStatus = true;
+    return;
+  }
+
+  if (function.empty()) {
+    LOG(ERROR) << "No analysis function defined";
+    mStatus = true;
+    return;
+  }
+
+  mAnalysis = GetFromMacro<CompressedAnalysis*>(filename, function, "CompressedAnalysis*", "compressed_analysis");
+  if (!mAnalysis) {
+    LOG(ERROR) << "Could not retrieve analysis from file: " << filename;
+    mStatus = true;
+    return;
+  }
+
+  mAnalysis->setDecoderCONET(conetmode);
+  mAnalysis->initialize();
+
+  auto finishFunction = [this]() {
+    LOG(INFO) << "CompressedBaseTask finish";
+    mAnalysis->finalize();
+  };
+  ic.services().get<CallbackService>().set(CallbackService::Id::Stop, finishFunction);
+}
+
+void CompressedAnalysisTask::run(ProcessingContext& pc)
+{
+
+  /** check status **/
+  if (mStatus) {
+    pc.services().get<ControlService>().readyToQuit(QuitRequest::Me);
+    return;
+  }
+
+  /** loop over inputs routes **/
+  for (auto iit = pc.inputs().begin(), iend = pc.inputs().end(); iit != iend; ++iit) {
+    if (!iit.isValid())
+      continue;
+
+    /** loop over input parts **/
+    for (auto const& ref : iit) {
+
+      const auto* headerIn = DataRefUtils::getHeader<o2::header::DataHeader*>(ref);
+      auto payloadIn = ref.payload;
+      auto payloadInSize = headerIn->payloadSize;
+
+      mAnalysis->setDecoderBuffer(payloadIn);
+      mAnalysis->setDecoderBufferSize(payloadInSize);
+      mAnalysis->run();
+    }
+  }
+}
+
+} // namespace tof
+} // namespace o2

